Documentation update

djesic opened the pull request:
Some things still need input:

> measurement_providers (System => Component / Machine. No more [SYSTEM])

I'm still seeing `system` providers on dev, can I get come more details on this please?

> Machine_ID and machine_description is now mandatory field and will be registered in DB on first run
> Introduced the cluster and machine key in config.yml

I did add these to the docs, but I feel like I don't know/understand enough about this to be able to document it correctly.
Could you please give me a couple of sentences about the intended use case for these?

Some things are gonna be done now:

> how to write your own metric provider

Before doing this, just wanted to double check if the current docs are still valid:

> Metric providers should follow the naming convention of: Where, What, How, Scope ie. `cpu.utilization.cgroup.container`

Find a place to document this:
> Please reverse the checking to rather block re.findall(r'(..|$|'|"|`|!)' than only allow ASCI (schema checker for usage scenario)

ArneTR commented:
Thanks for the detailed questions!

1. System is now for something that comes from a software source. Specifically the operating system (os).
2. The fields for machine_id and machine description are actually renamed and now under the `machine` key:
```
machine:
  id: 1
  description: "Development machine for testing"
```

If you think about our measurement we have a cluster of machines. The cluster is comprised of machine machines. Every machine has to "announce" what it is, by setting an ID and a description.
Whenever the machine does a measurement it will update the `machines` table on the cluster central DB with its info.
You can also check that behaviour in the `runner.py`.
The machine_id als tells the `jobs.py` to only select the correct machine for doing the measurement.
3, Metric providers should follow the naming convention of: Where, What, How, Scope ie.
=> This is extended now to:
Metric providers should follow the naming convention of: Where (what hardware / software source), What ("physical" unit) - This might be actualy two levels deep. for instance if you measure AC energy  we recommend doing `energy` -> `AC`. For network io only `io` is sufficient, How (HArdware / Software tool or resource that you are using. For instance procfs, RAPL etc), Scope (container, system, component, machine)

4. Find a place to document this
=> usage_scenario I would say

ribalba commented on content/en/docs/installation/installation-linux.md:
````diff
@@ -20,30 +20,30 @@ Please modify the commands accordingly.
 ```bash
 git clone https://github.com/green-coding-berlin/green-metrics-tool ~/green-metrics-tool && \
 cd ~/green-metrics-tool && \
-git submodule update --init && \
````
Why are you removing this?

djesic replied:
Submodules are now handled as part of the install script

ribalba commented on content/en/docs/installation/installation-linux.md:
```diff
@@ -20,30 +20,30 @@ Please modify the commands accordingly.
-git submodule update --init && \
 sudo apt update && \
 sudo apt upgrade -y && \
 sudo apt install make gcc python3 python3-pip libpq-dev -y && \
```
You could add `libglib2.0-dev` here so the error described above doesn't happen?

ribalba reviewed (changes requested):
Looks good. Please also add the
- Compare feature. What can we compare.
- Cluster

djesic commented:
@ribalba Unsure what to document for Cluster - is this the code? https://github.com/green-coding-berlin/green-metrics-tool/pull/210

ArneTR commented:
Didi has to do here probalby a more technical write-up, but for now please include just the information that we have the following machines in our cluster:

Fujitsu ESPRIMO P956 - Blue Angel compatible (Ubuntu 22.04)
Fujitsu TX1330 M2 - Single-Tenant Server (Ubuntu 22.04)
Quanta Leopard - 48-Thread Multi-Tenant Server - SoftAWERE compatible (Ubuntu 22.04)

ArneTR commented:
In the current version of the Green Metrics Tool we changed our measurement approach from just containing the data for the runtime of the software to also look at other stages of the lifecycle of software.

Software has not only to be run, it also has to be developed, tested, installed, booted and removed.

The Green Metrics Tool currently supports the following phases:

---

These phases originate from the idea to look at software from a lifecyclce perspective, similar to how [ISO 14001](link) and als the [Blue Angel](link) for software does.

ArneTR commented:
Please update the measuremet article with a screenshot of the phases and update the comparing article with quitem more details and a story as discussed

djesic commented:
@ArneTR @ribalba 
Ready for (hopefully) final review

Updated screenshots, added content about comparing, fixed markdown and spelling errors, added newlines for readability.

I've done some work on 'Philosophy & Methodology', but I would like some feedback on its content.
I have a feeling that it might be to be revised further.
